spark3/src/test/java/org/apache/iceberg/spark/sql/TestDeleteFrom.java

```diff
@@ -41,6 +43,7 @@ public void removeTables() {
 
   @Test
   public void testDeleteFromUnpartitionedTable() {
+    Assume.assumeFalse(Spark3VersionUtil.isSpark31());
```

Comment:
@aokolnychyi this test and the next fail in Spark 3.1. In https://github.com/apache/spark/blob/v3.1.1/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Strategy.scala#L250-L253, if `canDeleteWhere` returns false, there is no attempt to rewrite the query, an exception is just thrown. Is Spark supposed to rewrite the query if `SupportsDelete#canDeleteWhere` returns false? If so, this part hasn't been implemented in Spark yet.